components/HackneyDS/TreeView/index.js

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import { Container, HorizontalSeparator, Link } from '..';
+import { TreeViewItem } from './TreeViewItem';
+
+export const TreeView = ({ items, renderItem }) => {
+  const [openIds, modifyOpenIds] = useState([]);
+  const handleClick = (e, id) => {
+    e.preventDefault();
+    if (openIds.includes(id)) {
+      modifyOpenIds(openIds.filter((x) => x !== id));
+    } else {
+      // eslint-disable-next-line no-shadow
+      modifyOpenIds((openIds) => [...openIds, id]);
+    }
+  };
+  return items.map((item, index) => (
+    <Container key={item?.id} display="flex" flexDirection="column">
+      <TreeViewItem>
+        <Container
+          padding="24px 16px"
+          background="#FAFAFA"
+          display="flex"
+          flexDirection="column"
+          alignItems="flex-start"
+        >
+          {renderItem(item)}
+          {item?.children?.length > 0 && (
+            <>
+              <HorizontalSeparator height={10} />
+              <Link noVisited onClick={(e) => handleClick(e, item.id)}>
```

Comment:
I think we should make `Link` a `<button>` here instead of `<a>`, as obviously it will not be used with `href`

Therefore we don't need to use `e.preventDefault()` as well